tests/queries/0_stateless/03094_virtual_column_table_name.reference

```diff
@@ -0,0 +1,47 @@
+-- { echoOn }
+SELECT _table FROM d1;
+d1
+d1
+SELECT count(_table) FROM d1 WHERE _table = 'd1' GROUP BY _table;
+2
+SELECT _table, key, value FROM d1 WHERE value = 10;
+d1	1	10
+SELECT _table FROM d2;
+d2
+d2
+SELECT count(_table) FROM d2 WHERE _table = 'd2' GROUP BY _table;
+2
+SELECT _table, key, value FROM d2 WHERE value = 40;
+d2	4	40
+SELECT _table, value FROM d3 WHERE _table = 6;
+6	60
+SELECT _table FROM d4;
+d4
+d4
+SELECT count(_table) FROM d4 WHERE _table = 'd4' GROUP BY _table;
+2
+SELECT _table, key, value FROM d4 WHERE value = 40;
+d4	4	40
+SELECT _table FROM m1 ORDER BY _table ASC;
+d1
+d1
+d2
+d2
```

Comment:
Could you, please, explain why this behavior is a desired one and we do not want to see m1 here?

Reply:
Currently, the _table virtual column is only supported for tables with the Merge engine. This PR extends this virtual column to all tables.

Reply:
Let's make it always to be the inner-most table.

It will break the current behavior when a Merge table is created on top of more Merge tables, but it will be ok.